llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -389,7 +389,103 @@ InstructionCost RISCVTTIImpl::getShuffleCost(TTI::ShuffleKind Kind,
   // First, handle cases where having a fixed length vector enables us to
   // give a more accurate cost than falling back to generic scalable codegen.
   // TODO: Each of these cases hints at a modeling gap around scalable vectors.
-  if (isa<FixedVectorType>(Tp)) {
+  if (ST->hasVInstructions() && isa<FixedVectorType>(Tp)) {
+    MVT LegalVT = LT.second;
+    InstructionCost NumOfDests = InstructionCost::getInvalid();
```

Comment:
For readability, can you extract this into a helper function called costShuffleViaVRegSplitting?  That would make it easy to map the lowering code, and would allow use of early return to reduce indentation.